### PR TITLE
[IMP] deprecated-module: Add exception for openerp.osv.expression

### PR DESCRIPTION
**Pull request by moylop260**
Fix https://github.com/OCA/maintainer-quality-tools/issues/306 - `from openerp.osv import expression` is a valid import yet.

Output result example:
screen shot 2016-06-24 at 6 13 49 pm

- [x] [REF] tests: Add a dummy package `openerp.osv` to force `deprecated-module` [pylint#362](https://bitbucket.org/logilab/pylint/issues/362/bug-w0402-not-show-fails-if-module-not-is)
- [x] [REF] tests: Add `sys.path` to allow `import openerp.osv`

### CODE REVIEW

**Comment by moylop260** [comment]
@StefanRijnhart 
This is the feature requested

**Comment by StefanRijnhart** [comment]
Great work, thanks! :+1:

**Comment by pedrobaeza** [comment]
As there's a valid alias in Odoo (`from openerp.models import expression`), I see no need for this. References: 
8.0 - https://github.com/odoo/odoo/blob/ee3666237af305de7960103b66ab68afa9c894a9/openerp/models.py#L6215
9.0 - https://github.com/odoo/odoo/blob/44d23c371cebf888bc7438a4208f14a3ad789f99/openerp/models.py#L6282

**Comment by moylop260** [comment]
We can use transitive imports to bypass the error.
But we have people that dont like to use them.

Check the following comment of @gurneyalex

https://github.com/OCA/maintainer-quality-tools/issues/306#issuecomment-171592748

Other people want disable the check.

Other people want the check with transitive imports.

IMHO this change allow you use the best option for you without a problem of a false deprecated.

The expression is present from osv in latest master odoo version yet

If we search how odoo team `import` expression... we saw that they aren't use the transitive imports.
They are using osv.expression.

**Comment by pedrobaeza** [comment]
OK, but please include tests for the new method that doesn't lower the coverage.

**Comment by moylop260** [comment]
The test is included...
But in the main description I explained why don't work the coverage is down...

**Comment by moylop260** [comment]
FYI
Im workin on fix https://github.com/PyCQA/pylint/pull/954/files
But we will need wait a pylint release to have it in stable

This section just affect our tests because we don't have other tests that use this method.

Maybe if I force other test just to bypass the case...

Let me see

**Comment by moylop260** [comment]
FYI this PR is ready

**Comment by moylop260** [comment]
ping

**Comment by moylop260** [comment]
ping

**Comment by moylop260** [comment]
Well, 
IMHO We should take the 👍  of @guewen @yvaucher from https://github.com/OCA/maintainer-quality-tools/issues/306 comments too. Then I'll merge it with a big "sorry" (after 2 pings and 17 days ago without answers)

**Comment by StefanRijnhart** [comment]
Is the OCA dying?

**Comment by yvaucher** [comment]
@StefanRijnhart Maybe a part is on vacation? OCA is now quite big. Hard to follow everything.

**Comment by pedrobaeza** [comment]
These are very specific topics that not everyone can follow nor opinate. I was also quite busy these days to confirm that my comments were attended.

**Comment by dreispt** [comment]
@StefanRijnhart most Pylint & MQT PR are proposed by @moylop260 and then reviewed by me and @pedrobaeza 

In this case @pedrobaeza mentioned some concerns about this, and I think he has a point and was waiting for counter-arguments before I expressed an opinion.
That was enough to stall thhe PR.

**Comment by StefanRijnhart** [comment]
Oh, it's not just this particular PR. Most projects I visit are disfunctional.

**Comment by dreispt** [comment]
@StefanRijnhart Disfunctional in what way? Stalled PRs?

**Comment by gurneyalex** [comment]
@StefanRijnhart I've been really under the water during the 1st semester 2016. I'm recovering and slowly processing the github messages backlog that has stacked up high in my mailbox... 

May I suggest you raise the topic on the contributors mailing list, which is a better forum than github for such a broad discussion?

**Comment by StefanRijnhart** [comment]
On 08/03/2016 09:14 AM, Alexandre Fayolle wrote:
>
> @StefanRijnhart <https://github.com/StefanRijnhart> I've been really
> under the water during the 1st semester 2016. I'm recovering and
> slowly processing the github messages backlog that has stacked up high
> in my mailbox...
>

Ah, I was wondering about your absense. Sorry to hear you were not well.
Glad you are on the way up.

> May I suggest you raise the topic on the contributors mailing list,
> which is a better forum than github for such a broad discussion?
>

Yes, I should do that instead of trolling a random PR. Let me see if I
can find the time and the words.

**Comment by osvalr** [comment]
:+1:
